### PR TITLE
Remove game state modifications from load file

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Moves them into the higher level part of the function. This is important for the scenario list as we need to load a save without modifying state to list its properties.

### CODE REVIEW

File: src/OpenLoco/S5/S5.cpp [hunk 1]
```diff
@@ -453,33 +453,20 @@ namespace OpenLoco::S5
...
-                {
-                    objectInstalled = true;
-                }
-            }
-
-            if (objectInstalled)
-            {
-                ObjectManager::loadIndex();
+                auto unownedObjectData = fs.readChunk();
+                std::vector<uint8_t> objectData;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/S5/S5.cpp:461]: Might be nice to use `std::byte` here?
```suggestion
                std::vector<std::byte> objectData;
```

File: src/OpenLoco/S5/S5.h [hunk 2]
```diff
@@ -405,6 +405,7 @@ namespace OpenLoco::S5
         ObjectHeader requiredObjects[859];
         GameState gameState;
         std::vector<TileElement> tileElements;
+        std::vector<std::pair<ObjectHeader, std::vector<uint8_t>>> packedObjects;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/S5/S5.h:408]: And here?
```suggestion
        std::vector<std::pair<ObjectHeader, std::vector<std::byte>>> packedObjects;
```

**Comment by AaronVanGeffen** [comment]
> This is important for the scenario list as we need to load a save without modifying state to list its properties.

Is the current behaviour breaking anything (e.g. some scenarios aren't listed?) or is it more of an optimisation?

**Comment by duncanspumpkin** [comment]
> > This is important for the scenario list as we need to load a save without modifying state to list its properties.
> 
> Is the current behaviour breaking anything (e.g. some scenarios aren't listed?) or is it more of an optimisation?

I've not implemented it yet but when i looked at it i noticed we would need to make these changes (and a few others)

**Review by AaronVanGeffen — APPROVED** [review]
No objections to these changes; saving and loading appears to be working correctly. Not noting any divergence in produced saves either.

What do you think about my note on using `std::byte`?

File: src/OpenLoco/S5/S5.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/S5/S5.cpp:461]: I can look into that in a separate PR. Would be slightly messy as it needs changes in a few places.

**Pull request merged**